integrationtest/wildfly/src/test/java/org/hibernate/search/test/integration/jgroups/JGroupsDeploymentHelper.java

```diff
@@ -0,0 +1,129 @@
+import org.hibernate.search.test.integration.jms.MasterSlaveTestTemplate;
+import org.hibernate.search.test.integration.jms.controller.RegistrationController;
+import org.hibernate.search.test.integration.jms.model.RegisteredMember;
+import org.hibernate.search.test.integration.jms.util.RegistrationConfiguration;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.descriptor.api.Descriptors;
+import org.jboss.shrinkwrap.descriptor.api.persistence20.PersistenceDescriptor;
+import org.jboss.shrinkwrap.descriptor.api.persistence20.PersistenceUnit;
+import org.jboss.shrinkwrap.descriptor.api.persistence20.Properties;
+
+/**
+ * Create deployments using the JGroups backend
+ *
+ * @author Sanne Grinovero
+ */
+public class JGroupsDeploymentHelper {
+
+	private JGroupsDeploymentHelper() {
+		//not allowed
+	}
+
+	public static Archive<?> createMaster(String deploymentName, int refreshPeriodInSec, File tmpDir) throws Exception {
+		WebArchive master = baseArchive( deploymentName, masterPersistenceXml( deploymentName, refreshPeriodInSec, tmpDir ) );
+		addDependencyToSearchModule( master );
+		return master;
+	}
+
+	public static Archive<?> createSlave(String deploymentName, int refreshPeriodInSec, File tmpDir) throws Exception {
+		WebArchive slave = baseArchive( deploymentName, slavePersistenceXml( deploymentName, refreshPeriodInSec, tmpDir ) );
+		addDependencyToSearchModule( slave );
+		return slave;
+	}
+
+	private static WebArchive baseArchive(String name, PersistenceDescriptor unitDef) throws Exception {
+		WebArchive webArchive = ShrinkWrap
+				.create( WebArchive.class, name + ".war" )
+				.addClasses( RegistrationController.class, RegisteredMember.class, RegistrationConfiguration.class, MasterSlaveTestTemplate.class )
+				.addAsResource( new StringAsset( unitDef.exportAsString() ), "META-INF/persistence.xml" )
+				.addAsWebInfResource( EmptyAsset.INSTANCE, "beans.xml" );
+		return webArchive;
+	}
+
+	private static PersistenceDescriptor masterPersistenceXml(String name, int refreshPeriod, File tmpDir)
+			throws Exception {
+		return commonUnitDef( name, "filesystem-master", refreshPeriod, tmpDir )
+				.createProperty()
+					.name( "hibernate.search.default.worker.backend" )
+					.value( "jgroupsMaster" )
+					.up()
+				.up()
+			.up();
+	}
+
+	private static PersistenceDescriptor slavePersistenceXml(String name, int refreshPeriod, File tmpDir)
+			throws Exception {
+		return commonUnitDef( name, "filesystem-slave", refreshPeriod, tmpDir )
+					.createProperty()
+						.name( "hibernate.search.default.worker.backend" )
+						.value( "jgroupsSlave" )
+						.up()
+					.up()
+				.up();
+	}
+
+	private static Properties<PersistenceUnit<PersistenceDescriptor>> commonUnitDef(
```

Comment:
I'm not against using the files, just that we would have three of them for
each master / slave test? But right they would probably be more readable
:-)

On Tue, 30 Jun 2015 08:24 Gunnar Morling <notifications@github.com> wrote:

> In
> integrationtest/wildfly/src/test/java/org/hibernate/search/test/integration/jgroups/JGroupsDeploymentHelper.java
> <https://github.com/hibernate/hibernate-search/pull/846#discussion_r33546508>
> :
>
> > +				.up()
> > +			.up();
> > +	}
> > +
> > +	private static PersistenceDescriptor slavePersistenceXml(String name, int refreshPeriod, File tmpDir)
> > +			throws Exception {
> > +		return commonUnitDef( name, "filesystem-slave", refreshPeriod, tmpDir )
> > +					.createProperty()
> > +						.name( "hibernate.search.default.worker.backend" )
> > +						.value( "jgroupsSlave" )
> > +						.up()
> > +					.up()
> > +				.up();
> > +	}
> > +
> > +	private static Properties<PersistenceUnit<PersistenceDescriptor>> commonUnitDef(
>
> Out of curiosity, do you actually feel that this style of descriptor
> declaration is advantageous? Granted, it may help preventing some
> duplications, but personally I would be willing to accept those for the
> sake of a better readable actual persistence.xml file (prepped with
> variables via resource filtering as needed). Which also can be easily used
> as template for other examples or hints to users etc.
>
> No need to change it, just interested in your opinion, maybe there are
> advantages I miss.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hibernate/hibernate-search/pull/846/files#r33546508>.
>